### PR TITLE
legacymarket: allow migration of liquidatable accounts

### PR DESCRIPTION
**Pull request by dbeal-eth**
we remove the code that allows for the legacymarket to "liquidate" accounts, and instead allow even liquidatable accounts to be migrated so that the v3 liquidation system can be used instead.

this change works by selectively removing a restriction on the v3 core system which prevented the assignment of debt which would cause an account to become liquidatable, a tactic that would allow a market to mint effectively unlimited debt.

### CODE REVIEW

File: markets/legacy-market/contracts/LegacyMarket.sol [hunk 1]
```diff
@@ -222,25 +224,7 @@ contract LegacyMarket is ILegacyMarket, Ownable, UUPSImplementation, IMarket, IE
         uint256 cratio = (collateralMigrated * v3System.getCollateralPrice(address(oldSynthetix))) /
             debtValueMigrated;
 
-        // if the account needs to be liquidated, liquidate it here by unlocking the debt to all accounts and moving to a
         if (
-            cratio < v3System.getCollateralConfiguration(address(oldSynthetix)).liquidationRatioD18
```
Comment (kaleb-keny) [inline markets/legacy-market/contracts/LegacyMarket.sol:227]: If we delete this line how does the liquidation workflow work? I mean how can one force a v2x liquidation, without hitting reverts due to not owning the account when calling migrate?

**Review by kaleb-keny — APPROVED** [review]

File: markets/legacy-market/cannonfile.toml [hunk 2]
```diff
@@ -103,6 +96,18 @@ extra.marketId.event = "MarketRegistered"
 extra.marketId.arg = 1
 depends = ["invoke.setFeatureFlagMarket", "invoke.setSystemAddresses"]
 
+# tell the v3 system its ok for this market to assume unlimited debt (to allow underwarter stakers to be converted and then liquidated)
+[invoke.setConfigAssociateUnlimitedDebt]
+target = ["v3.CoreProxy"]
+fromCall.func = "owner"
+func = "setConfig"
+args = [
+    "<%= keccak256(defaultAbiCoder.encode(['bytes32', 'uint256'], [formatBytes32String('associateDebtRatioExclusion'), settings.marketId])) %>",
```
Comment (barrasso) [inline markets/legacy-market/cannonfile.toml:105]: should be `settings.market_id`

File: utils/core-contracts/contracts/utils/RevertUtil.sol [hunk 3]
```diff
@@ -0,0 +1,15 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.11 <0.9.0;
+
+/*
+		Utilities related to reverts and error handling
+*/
+
+library RevertUtil {
```
Comment (barrasso) [inline utils/core-contracts/contracts/utils/RevertUtil.sol:8]: Did this get added by mistake? Harmless, but seems out of scope/unused

**Review by barrasso — CHANGES_REQUESTED** [review]

File: utils/core-contracts/contracts/utils/RevertUtil.sol (hunk 3 above)
Comment (dbeal-eth) [reply utils/core-contracts/contracts/utils/RevertUtil.sol:8]: ah whoops. I think I checked it in because I thought I forgot to include the file. probably goes with a different PR :/

**Review by barrasso — APPROVED** [review]
